README.md

```diff
@@ -316,6 +316,21 @@ record :with_date_array
 end
 ```
 
+### User-defined Types
```

Comment:
Do you think this terminology will be confused with named types? Would something like `type_alias` or `typo_macro` be clearer?